dind/Dockerfile

```diff
@@ -1,26 +1,35 @@
 ARG DOCKER_VERSION
```

Comment:
This needs to go after the build multistage block. I have experienced problems before with ARGs going out of scope and causing unexpected behaviors.

Reply:
Can't get it to build that way:
```
...
Step 7/20 : ARG DOCKER_VERSION
 ---> Using cache
 ---> 341f9cfeb759
Step 8/20 : FROM docker:${DOCKER_VERSION}-dind-rootless
invalid reference format
```

Reply:
Hmmm, let me take a look at that.

Reply:
I was actually mistaken, the problem I had experienced was regarding ARGs used in the runtime of multi-stage builds and it doesn't apply to the FROM instruction.

https://docs.docker.com/engine/reference/builder/#understand-how-arg-and-from-interact